caikit/core/exceptions/caikit_core_exception.py

```diff
@@ -36,3 +37,4 @@ class CaikitCoreException(Exception):
     def __init__(self, status_code: CaikitCoreStatusCode, message: str) -> None:
         self.status_code = status_code
         self.message = message
+        self.id = uuid.uuid4().hex
```

Comment:
Why is this needed?

Reply:
I wanted to match the format of [CaikitRuntimeException](https://github.com/caikit/caikit/blob/70685013943c63b15f57b3526eb1148675185705/caikit/runtime/types/caikit_runtime_exception.py#L44) so they could reuse the same error handlers